corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -249,6 +249,10 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                     oldValue = $input.val(),
                     parts = _.map(oldValue.split(separator), function (v) { return moment(v); }),
                     newValue = '';
+                // conditions where only one valid date is typed in rather than a range
+                if (parts.length === 1 && function (parts) { return parts.isValid(); }) {
```

Comment:
```suggestion
                if (parts.length === 1 && function (parts) { return parts.isValid(); }()) {
```

Reply:
ope thank you-- that brought up too that parts should be checked the same as the other conditional so I modified to check if the parts are a valid length before the length check